sky/jobs/recovery_strategy.py

```diff
@@ -289,26 +301,53 @@ def _launch(self,
                         state.set_restarting(self.job_id, self.task_id,
                                              recovery)
                     try:
-                        usage_lib.messages.usage.set_internal()
-                        # Detach setup, so that the setup failure can be
-                        # detected by the controller process (job_status ->
-                        # FAILED_SETUP).
-                        execution.launch(
-                            self.dag,
-                            cluster_name=self.cluster_name,
-                            # We expect to tear down the cluster as soon as the
-                            # job is finished. However, in case the controller
-                            # dies, set autodown to try and avoid a resource
-                            # leak.
-                            idle_minutes_to_autostop=_AUTODOWN_MINUTES,
-                            down=True,
-                            _is_launched_by_jobs_controller=True)
-                        logger.info('Managed job cluster launched.')
+                        if not state.force_no_postgres():
+                            usage_lib.messages.usage.set_internal()
+                            # Detach setup, so that the setup failure can be
+                            # detected by the controller process (job_status ->
+                            # FAILED_SETUP).
+                            self._logger.info('Launching job with sdk')
+                            request_id = sdk.launch(
+                                self.dag,
+                                cluster_name=self.cluster_name,
+                                idle_minutes_to_autostop=_AUTODOWN_MINUTES,
+                                down=True,
+                                _is_launched_by_jobs_controller=True)
+                            # would be nice to have the async version of this,
+                            # however, scheduled_launch uses scheduler
+                            # functions that acquire filelocks, no async file
+                            # lock exists so it would have to be run in a
+                            # coroutine anyway.
+                            log_file = _get_logger_file(self._logger)
+                            try:
+                                if log_file is None:
+                                    # will get caught by the except block below
+                                    raise OSError('Log file is None')
+                                with open(log_file, 'a', encoding='utf-8') as f:
+                                    sdk.stream_and_get(request_id,
+                                                       output_stream=f)
+                            except OSError as e:
+                                self._logger.error(
+                                    f'Failed to stream logs: {e}')
+                                sdk.get(request_id)
```

Comment:
I don't love this logging stuff. It randomly stop working if we change how the logger is set up and will be pretty hard to figure out what's going on. Maybe it's okay if we add a comment to the logging setup, mentioning that this code will break if we don't use a file logger.